### PR TITLE
[5.1] Fix EncryptCookies middleware not disabling encryption for queued cookies defined in the $except property

### PR DESCRIPTION
**Pull request by LaurentEsc**
PR related to issue #9317.

### CODE REVIEW

File: src/Illuminate/Cookie/Middleware/EncryptCookies.php [hunk 1]
```diff
@@ -121,7 +121,7 @@ protected function decryptArray(array $cookie)
     protected function encrypt(Response $response)
     {
         foreach ($response->headers->getCookies() as $key => $cookie) {
```
Comment (GrahamCampbell) [inline src/Illuminate/Cookie/Middleware/EncryptCookies.php]: You need to delete `$key => ` from here

**Comment by franzliedke** [comment]
Should we do the same for the `decrypt()` function?

Also, you were saying this might be related to queued cookies (they probably had different keys). Should we consider using the names as keys when accessing the queued cookies?

**Comment by GrahamCampbell** [comment]
Could you include tests please?

**Comment by franzliedke** [comment]
There are currently no tests for this middleware...

**Comment by GrahamCampbell** [comment]
Yeh, we need some....

File: src/Illuminate/Cookie/Middleware/EncryptCookies.php (hunk 1 above)
Comment (LaurentEsc) [inline src/Illuminate/Cookie/Middleware/EncryptCookies.php]: Done

**Comment by LaurentEsc** [comment]
I have included a test for the Middleware encryption. Please review, I am not very comfortable with Middleware testing.

File: tests/Cookie/Middleware/EncryptCookiesTest.php [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+            return new Encrypter(str_repeat('a', 16));
+        });
+
+        $this->router = new Router(new Illuminate\Events\Dispatcher, $container);
+    }
+    public function testSetCookieEncryption()
+    {
+        $this->router->get($this->setCookiePath, [
+            'middleware' => 'EncryptCookiesTestMiddleware',
+            'uses'=>'EncryptCookiesTestController@setCookies'
```
Comment (GrahamCampbell) [inline tests/Cookie/Middleware/EncryptCookiesTest.php]: cs

File: tests/Cookie/Middleware/EncryptCookiesTest.php [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+        $this->assertEquals('encrypted_cookie', $cookies[0]->getName());
+        $this->assertNotEquals('value', $cookies[0]->getValue());
+        $this->assertEquals('unencrypted_cookie', $cookies[1]->getName());
+        $this->assertEquals('value', $cookies[1]->getValue());
+    }
+    public function testQueuedCookieEncryption()
+    {
+        $this->router->get($this->queueCookiePath, [
+            'middleware' => ['EncryptCookiesTestMiddleware', 'AddQueuedCookiesToResponseTestMiddleware'],
+            'uses'=>'EncryptCookiesTestController@queueCookies'
```
Comment (GrahamCampbell) [inline tests/Cookie/Middleware/EncryptCookiesTest.php]: cs

**Comment by LaurentEsc** [comment]
cs fixed

File: tests/Cookie/Middleware/EncryptCookiesTest.php [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+        $cookies = $response->headers->getCookies();
+        $this->assertEquals(2, count($cookies));
+        $this->assertEquals('encrypted_cookie', $cookies[0]->getName());
+        $this->assertNotEquals('value', $cookies[0]->getValue());
+        $this->assertEquals('unencrypted_cookie', $cookies[1]->getName());
+        $this->assertEquals('value', $cookies[1]->getValue());
+    }
+}
+
+class EncryptCookiesTestController extends Illuminate\Routing\Controller
```
Comment (GrahamCampbell) [inline tests/Cookie/Middleware/EncryptCookiesTest.php]: Please import all classes you use

**Pull request merged**